device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/IotHubConnectionString.java

```diff
@@ -141,7 +141,7 @@ public IotHubConnectionString(String hostName, String deviceId,
                                   String sharedAccessKey, String sharedAccessToken)
             throws IllegalArgumentException, URISyntaxException
     {
-        this(hostName, deviceId, sharedAccessKey, sharedAccessToken, "");
+        this(hostName, deviceId, sharedAccessKey, sharedAccessToken, null);
```

Comment:
Within the mqtt layer, there is a check when subscribing to c2d messaging where we do not subscribe to c2d if connecting to an edgehub. That check was just if the gatewayhostname (argument 5 above) is null or not. Since it was empty, mqtt did not subscribe to c2d messaging.